class.jetpack.php

```diff
@@ -4749,6 +4762,43 @@ function authenticate_jetpack( $user, $username, $password ) {
 		return new WP_User( $token_details['user_id'] );
 	}
 
+	// Authenticates requests from Jetpack server to WP API endpoints.
+	// Uses the existing XMLRPC oAuth implementation.
+	function wp_rest_authenticate( $error ) {
+		if ( is_wp_error( $error ) ) {
+			// A previous authentication method failed.
+			return $error;
+		}
+
+		if ( ! isset( $_GET['token'] ) && ! isset( $_GET['signature'] ) ) {
+			// Nothing to do for this authentication method.
+			return $error;
+		}
+
+		$verified = $this->verify_xml_rpc_signature();
+
+		if ( is_wp_error( $verified ) ) {
+			return $verified;
+		}
+
+		if (
+			false === $verified ||
+			! isset( $verified['type'] ) ||
```

Comment:
Doesn't matter - this can only ever be `blog` or `user`.  See `$token_type` [in `verify_xml_rpc_signature`](https://github.com/Automattic/jetpack/blob/a14a731/class.jetpack.php#L4630).